templates/acronym_identification/templates.yaml

```diff
@@ -0,0 +1,161 @@
+      \ %}\n  {% if labels[label_idx] == 0 %}{# Long Beginning #}\n    {% if exp_string.value!=''\
+      \ and exp_string.value not in exp_abbr_dict.value.keys() %}{# Some string already\
+      \ present #}\n       {% set _dummy = exp_abbr_dict.value.update({exp_string.value:''})\
+      \ %}{# Discard this string as a new long string is coming #} \n    {% endif\
+      \ %}\n    {% set exp_string.value = tokens[label_idx] %}{# Create new long string\
+      \ #}\n  {% elif labels[label_idx] == 1 %}{# Short Beginning #}\n    {% set abbr_string.value\
+      \ = tokens[label_idx] %}{# Create new short string #}\n  {% elif labels[label_idx]\
+      \ == 2 %}{# Long Intermediate #}\n    {% set exp_string.value = exp_string.value+'\
+      \ '+tokens[label_idx] %}{# Update existing string #}\n  {% elif labels[label_idx]\
+      \ == 3 %}{# Short Intermediate #}\n    {% set abbr_string.value = abbr_string.value+tokens[label_idx]\
+      \ %}{# Update existing string #}\n  {% else %}{# Other #}\n    {# Both non-empty,\
+      \ and first characters match #}\n    {% if abbr_string.value!='' and exp_string.value!=''\
+      \ and exp_string.value.split()[0][0]|lower in abbr_string.value|lower and exp_string.value.split()[-1][0]|lower\
+      \ in abbr_string.value|lower%}\n      {# Update both the dictionaries #}\n \
+      \     {% set _dummy = exp_abbr_dict.value.update({exp_string.value:abbr_string.value})\
+      \ %}\n      {# Empty both the strings #}\n      {% set abbr_string.value= ''\
+      \ %}\n      {% set exp_string.value= '' %}\n    {% endif %}\n  {% endif %}\n\
+      {% endfor %}\n{# Both non-empty, and first characters match #}\n{% if abbr_string.value!=''\
+      \ and exp_string.value!='' %}\n  {% if exp_string.value.split()[0][0]|lower\
+      \ in abbr_string.value|lower and exp_string.value.split()[-1][0]|lower in abbr_string.value|lower\
+      \ %}\n    {# Update the dictionary #}\n    {% set _dummy = exp_abbr_dict.value.update({exp_string.value:abbr_string.value})\
+      \ %}\n  {% elif exp_abbr_dict.value.items()|length==0 %}\n    {% set _dummy\
+      \ = exp_abbr_dict.value.update({exp_string.value:abbr_string.value}) %}\n  {%\
+      \ endif %}\n{% else %}\n  {% if exp_string.value!='' %}\n    {% if exp_string.value\
+      \ not in exp_abbr_dict.value.keys() %}\n      {% set _dummy = exp_abbr_dict.value.update({exp_string.value:''})\
+      \ %}\n    {% endif %}\n  {% endif %}\n{% endif %}\n{% if exp_abbr_dict.value.items()|length!=0\
+      \ %}\n{% set random_exp = exp_abbr_dict.value.keys()|list|choice %}\nGiven the\
+      \ following tokens, find the abbreviation for {{random_exp}}:\n \n{{tokens|join('\
+      \ ')}}\n|||\n{% if random_exp in exp_abbr_dict.value.keys() and exp_abbr_dict.value[random_exp]!=''\
+      \ %}\n{{exp_abbr_dict.value[random_exp]}}\n{% else %}\nUnclear\n{% endif %}\n\
+      {% else %}\n|||\n{% endif %}"
```

Comment:
bring the `|||` out of the your if condition. if the template is blank, it should return an empty string `` and then through the `.split("|||")` it should be `[""]`

Reply:
seems like it's persisting:
<img width="669" alt="Capture d’écran 2021-06-17 à 2 27 37 PM" src="https://user-images.githubusercontent.com/16107619/122455373-50978880-cf7a-11eb-9201-b69c1d4ff89e.png">

It should literally be displaying `∅∅∅ *Blank result*` (see https://github.com/bigscience-workshop/promptsource/blob/main/promptsource/promptsource.py#L303)
and I think right now there are a lot of "\n" in what's being returned...